app/models/discussion/discussion.rb

```diff
@@ -163,6 +163,12 @@ def post_destroyed(post, decrement=true)
       last_post: visible_posts.last,
       replied_by_id: visible_posts.last.try.user_id,
       replied_at: visible_posts.last.try.updated_at )
+    if post.private?
+      post_reciever = post.discussion.user_talking_to(post.user)
+      private_notices = PrivateMessageNotice.for_user(post_reciever).where(dismissed: false)
+      post_data = { from: post.user.name, message: post.body_html }
+      private_notices.detect { |n| n.data == post_data }.try(:dismiss!)
```

Comment:
I think there's a better way to do this. Imagine the message body is edited later on. Then this would not work anymore. So I would prefer this:
```ruby
PrivateMessageNotice.where(noticeable_type: post.type, noticable_id: post.id, dismissed: false)
```
I'm not sure if we actually need to verify the recipient. I would argue that any notice about a deleted message can be dismissed. Right now it probably does not make any difference because we only notify the recipient of the message.